core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -598,6 +598,49 @@ class DAGSchedulerSuite
     assertDataStructuresEmpty()
   }
 
+  test("run with ShuffleMapStage retry") {
+    val firstRDD = new MyRDD(sc, 3, Nil)
+    val firstShuffleDep = new ShuffleDependency(firstRDD, null)
+    val firstShuffleId = firstShuffleDep.shuffleId
+    val shuffleMapRdd = new MyRDD(sc, 3, List(firstShuffleDep))
+    val shuffleDep = new ShuffleDependency(shuffleMapRdd, null)
+    val reduceRdd = new MyRDD(sc, 1, List(shuffleDep))
+    submit(reduceRdd, Array(0))
+
+    complete(taskSets(0), Seq(
+      (Success, makeMapStatus("hostB", shuffleMapRdd.partitions.size)),
+      (Success, makeMapStatus("hostB", shuffleMapRdd.partitions.size)),
+      (Success, makeMapStatus("hostC", shuffleMapRdd.partitions.size))
+    ))
+
+    complete(taskSets(1), Seq(
+      (Success, makeMapStatus("hostA", reduceRdd.partitions.size)),
+      (Success, makeMapStatus("hostB", reduceRdd.partitions.size))
+    ))
+    runEvent(ExecutorLost("exec-hostA"))
+    // Resubmit already succcessd in hostA task
+    runEvent(CompletionEvent(taskSets(1).tasks(0), Resubmitted,
+      null, null, createFakeTaskInfo(), null))
+
+    // Cause mapOutTracker remove hostA outputs for taskset(0).
+    // Task that resubmitted will fetch matadata failed.
+    runEvent(CompletionEvent(taskSets(1).tasks(0),
+      FetchFailed(null, firstShuffleId, -1, 0, "Fetch matadata failed"),
+      null, null, createFakeTaskInfo(), null))
+    // FetchFailed cause resubmit failed Stages.
+    scheduler.resubmitFailedStages()
+
+    runEvent(CompletionEvent(taskSets(1).tasks(0), Success,
+      makeMapStatus("hostC", reduceRdd.partitions.size), null, createFakeTaskInfo(), null))
+    runEvent(CompletionEvent(taskSets(1).tasks(2), Success,
+      makeMapStatus("hostC", reduceRdd.partitions.size), null, createFakeTaskInfo(), null))
```

Comment:
3ks, I will refine that test... A very long time without updating that patch... so some logical slip from one's memory